pkg/services/ngalert/alert_query.go

```diff
@@ -0,0 +1,238 @@
+package ngalert
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"time"
+
+	"github.com/grafana/grafana-plugin-sdk-go/backend"
+)
+
+const defaultMaxDataPoints float64 = 100
+const defaultIntervalMs float64 = 1000
+const defaultExprDatasourceID = -100
+
+type duration time.Duration
+
+func (d duration) MarshalJSON() ([]byte, error) {
+	return json.Marshal(time.Duration(d).String())
+}
+
+func (d *duration) UnmarshalJSON(b []byte) error {
+	var v interface{}
+	if err := json.Unmarshal(b, &v); err != nil {
+		return err
+	}
+	switch value := v.(type) {
+	case float64:
+		*d = duration(time.Duration(value))
```

Comment:
I think this would make it nanoseconds ... why would we have a string or float?

Reply:
I will change it to accept only float64 (seconds) for now and we can reconsider changing it if a string representation will be more convenient in the future.